pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/admin/UserAdminService.java

```diff
@@ -0,0 +1,844 @@
+
+    private static final String SQL_INSERT_USER_ROLE =
+            "INSERT INTO access_user_role_group SET user_id = ?, role_id = ?, group_id = ?";
+
+    private static final String SQL_DELETE_USER_ROLES =
+            "DELETE FROM access_user_role_group WHERE user_id = ?";
+
+    private static final String SQL_DELETE_GROUP_ROLE =
+            "DELETE FROM ddp_group_role WHERE group_role_id = ?";
+
+    private static final String SQL_DELETE_USER_ROLE =
+            "DELETE FROM access_user_role_group WHERE user_id = ? AND role_id = ? AND group_id = ?";
+
+    private static final String SQL_INSERT_GROUP =
+            "INSERT INTO ddp_group SET name = ?";
+
+    private static final String SQL_DELETE_GROUP =
+            "DELETE FROM ddp_group WHERE group_id = ?";
+
+    private static final String SQL_INSERT_GROUP_ROLE =
+            "INSERT INTO ddp_group_role SET group_id = ?, role_id = ?, admin_role_id = ?";
+
+    private static final String SQL_SELECT_GROUP_FOR_REALM =
+            "SELECT dg.group_id, dg.name from ddp_group dg "
+                    + "JOIN ddp_instance_group dig on dig.ddp_group_id = dg.group_id "
+                    + "JOIN ddp_instance di on di.ddp_instance_id = dig.ddp_instance_id "
+                    + "WHERE di.instance_name = ?";
+
+    public UserAdminService(String operatorId, String studyGroup) {
+        this.operatorId = operatorId;
+        this.studyGroup = studyGroup;
+    }
+
+    /**
+     * Get study roles that operator can administer
+     */
+    public StudyRoleResponse getStudyRoles() {
+        int groupId = verifyStudyGroup(studyGroup);
+        Map<String, RoleInfo> studyRoles = getAdminRoles(groupId);
+
+        StudyRoleResponse res = new StudyRoleResponse();
+        for (var entry: studyRoles.entrySet()) {
+            RoleInfo roleInfo = entry.getValue();
+            res.addRole(new StudyRoleResponse.Role(roleInfo.getName(), roleInfo.getDisplayText()));
+        }
+        return res;
+    }
+
+    /**
+     * Add roles for list of users
+     */
+    public void addUserRoles(UserRoleRequest req) {
+        validateRoleRequest(req);
+
+        int groupId = verifyStudyGroup(studyGroup);
+        Map<String, RoleInfo> studyRoles = getAdminRoles(groupId);
+        List<String> roleNames = req.getRoles();
+        validateRoles(roleNames, studyRoles.keySet());
+
+        for (String userEmail: req.getUsers()) {
+            if (StringUtils.isBlank(userEmail)) {
+                throw new DSMBadRequestException("Invalid user email: blank");
+            }
+            addRoles(userEmail, roleNames, groupId, studyRoles);
+        }
+    }
+
+    protected void addRoles(String email, List<String> roles, int groupId, Map<String, RoleInfo> studyRoles) {
+        int userId = verifyUserByEmail(email, groupId);
+
+        for (String role : roles) {
+            if (StringUtils.isBlank(role)) {
+                throw new DsmInternalError("Invalid role: blank");
```

Comment:
Shouldn't we check for this before the for? In order to avoid half of the roles getting inserted before finding a blank role in the middle of `roles`

Reply:
Yes! I actually improved that in my follow-up branch (for editing user info). 
Is it okay if I leave that code in the other branch (it will be ready to merge once this is approved), or would you prefer to have it in this branch?